CRM/Core/EntityTokens.php

```diff
@@ -16,7 +16,14 @@
 use Civi\Token\TokenProcessor;
 
 /**
- * Class CRM_Core_EntityTokens
+ * Generic base-class which loads tokens via APIv4.
+ *
+ * To write a subclass:
+ *
+ * - (MUST) Implement getEntityName() and getApiEntityName()
+ * - (MUST) Implement getApiTokens()
+ * - (MAY) Implement getAliasTokens()
```

Comment:
I think we should offer 'getDeprecatedTokens` not `getAliasTokens` - we are going to need  a way to deprecate tokens but we only want to have one 'right' token for each thing.